### PR TITLE
refactor: Implement proposed fixes for MapData [skip ci]

### PR DESCRIPTION
**Pull request by magicus**
This is re-created from #2375

### CODE REVIEW

File: common/src/main/java/com/wynntils/utils/StringUtils.java [hunk 1]
```diff
@@ -171,6 +171,15 @@ public static ByteBuffer decodeBase64(String base64) {
...
     }
 
+    public static int tryParseInt(String value, int defaultValue) {
+        if (value == null) return defaultValue;
+        try {
+            return Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            return defaultValue;
+        }
+    }
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/utils/StringUtils.java]: This method is unused now.

**Review by kristofbolyai — APPROVED** [review]
Otherwise, every change looks good.

**Comment by kristofbolyai** [comment]
I added skip-ci. Or was there a reason, not to @magicus?

**Comment by magicus** [comment]
No, not really.

**Comment by kristofbolyai** [comment]
Should we merge this PR, or is it still in the works?

**Comment by magicus** [comment]
I'm still working on it. I think I'm going to split this into two parts, one which fixes visibility and one that fixes the level requirements.

**Comment by magicus** [comment]
This is partially replaced by https://github.com/Wynntils/Artemis/pull/2493. The label changes will be presented in an upcoming PR.

**Pull request closed without merging**